ui.frontend/src/view/FormField.js

```diff
@@ -57,23 +58,32 @@ export default class FormField {
        throw new Error ("Not Implemented");
     }
 
-    getTagName() {
-        throw new Error ("Not Implemented");
-    }
-
     setValue(value) {
        throw new Error("Not implemented");
     }
 
+    setFocus() {
+        throw new Error("Not implemented");
+    }
+
+    setParent(parentView) {
+        this.parentView = parentView;
+        if (this.parentView.addChild) {
+            this.parentView.addChild(this);
+        }
+    }
+
+    //link class of state, single function to handle various properties of state
+    setState(state) {
+        throw new Error("Not implemented");
+    }
+
     subscribe() {
+        this.setState(this._model);
         this._model.subscribe((action) => {
             let state = action.target.getState();
-            if (!state.valid) {
-                alert(state.errorMessage);
-                this.setValue(null);
-            } else {
-                this.setValue(state.value);
-            }
+            //action.changes[{prop, newValue, currValue}]
```

Comment:
@vdua I couldn't find an API to access prop, newValue, currValue. Can you point me to that API, it will optimize HTML attribute setting.